provider.go

```diff
@@ -88,8 +90,122 @@ func (EnvProvider) Priority() int {
 	return 2
 }
 
-func (EnvProvider) FormatKey(key string) (env string) {
-	env = strings.ToUpper(key)
+// FormatEnvKey transforms configuration keys to environment variable format.
+// Examples: "database.url" -> "DATABASE_URL", "api-key" -> "API_KEY"
+func FormatEnvKey(key string) string {
+	env := strings.ToUpper(key)
 	env = strings.Replace(env, ".", "_", -1)
 	return strings.Replace(env, "-", "_", -1)
 }
+
+func (EnvProvider) FormatKey(key string) (env string) {
+	return FormatEnvKey(key)
+}
+
+// A Provider that implements the repository.Provider interface for dotenv files.
+type DotenvProvider struct {
+	vars map[string]string
+}
+
+// NewDotenvProvider creates a provider that loads environment variables from a .env file
+// in the current directory. If the file doesn't exist, the provider will be empty but still functional.
+func NewDotenvProvider() *DotenvProvider {
+	return NewDotenvProviderWithPath(".env")
+}
+
+// NewDotenvProviderWithPath creates a provider that loads environment variables from the specified .env file.
+// If the file doesn't exist, the provider will be empty but still functional.
+func NewDotenvProviderWithPath(path string) *DotenvProvider {
+	p := &DotenvProvider{
+		vars: make(map[string]string),
+	}
+
+	// Make path absolute to avoid issues with working directory changes
+	if !filepath.IsAbs(path) {
+		if abs, err := filepath.Abs(path); err == nil {
+			path = abs
+		}
+	}
+
+	p.loadFile(path)
+	return p
+}
+
+// loadFile loads variables from the specified dotenv file.
+func (p *DotenvProvider) loadFile(path string) {
+	file, err := os.Open(path)
+	if err != nil {
+		// File doesn't exist or can't be opened, but that's okay
+		return
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+
+		// Skip empty lines and comments
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		// Parse KEY=VALUE format
+		key, value, found := strings.Cut(line, "=")
+		if !found {
+			continue
+		}
+
+		key = strings.TrimSpace(key)
+		value = strings.TrimSpace(value)
+
+		// Parse quoted values with basic unquoting
+		value = p.unquoteValue(value)
+
+		p.vars[key] = value
+	}
+}
+
+// unquoteValue removes surrounding quotes and handles basic escape sequences.
+// Note: This is a simplified implementation. For full shell-like quoting,
+// consider using strconv.Unquote or a more sophisticated parser.
+func (p *DotenvProvider) unquoteValue(value string) string {
+	if len(value) < 2 {
+		return value
+	}
+
+	// Handle double quotes
+	if value[0] == '"' && value[len(value)-1] == '"' {
+		inner := value[1 : len(value)-1]
+		// Basic escape sequence handling for double quotes
+		inner = strings.ReplaceAll(inner, `\"`, `"`)
+		inner = strings.ReplaceAll(inner, `\\`, `\`)
+		inner = strings.ReplaceAll(inner, `\n`, "\n")
+		inner = strings.ReplaceAll(inner, `\t`, "\t")
+		return inner
+	}
```

Comment:
You have to return here, otherwise

FOO="'foo'"

Won't be computed as `foo`, instead of `'foo'`

Please add a test about it